### PR TITLE
plank 1.2 (new formula)

### PR DESCRIPTION
**Pull request by rahul-malik**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/plank.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Plank < Formula
+  plank_version = "v1.2"
```
Comment (DomT4) [inline Formula/plank.rb]: This format combined with using it in the URL is unsupported.

File: Formula/plank.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  desc "Framework for generating immutable model objects"
+  homepage "https://pinterest.github.io/plank/"
+  url "https://github.com/pinterest/plank.git", :tag => plank_version
```
Comment (DomT4) [inline Formula/plank.rb]:
```
https://github.com/pinterest/plank/archive/v1.2.tar.gz
```
Unless there's a good reason to build from a git checkout.

File: Formula/plank.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  head "https://github.com/pinterest/plank.git"
+
+  depends_on :xcode => ["9.0", :build]
+  depends_on "kylef/formulae/swiftenv" => [:build]
```
Comment (DomT4) [inline Formula/plank.rb]: This is not permitted in `homebrew/core`. Will either have to be added as a new formula or vendored in as a resource, but I'm not sure what you're aiming for here with `swiftenv`? Kinda unusual to use a version manager as a build tool.

File: Formula/plank.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+  def install
+    system "unset CC; make archive"
```
Comment (DomT4) [inline Formula/plank.rb]: `unset CC; ` - Remove this.

**Comment by rahul-malik** [comment]
@DomT4 @fxcoudert - Thanks for your feedback and I've updated the PR for your comments. Please let me know if there is anything else I need to do

File: Formula/plank.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+  depends_on :xcode => ["9.0", :build]
+
+  def install
+    system "make", "archive"
+    bin.install ".build/release/plank"
+  end
+
+  test do
+    system "#{bin}/plank", "--help"
```
Comment (DomT4) [inline Formula/plank.rb]: Going to need some form of functionality testing here. Something like https://github.com/pinterest/plank#defining-a-schema and the blob below & then asserting to check those generated files exist seems reasonable.

**Comment by rahul-malik** [comment]
@DomT4 - Went through and added functionality tests and worked through any remaining issues reported by Jenkins CI. Let me know if there is anything else I need to do before this can be merged

File: Formula/plank.rb [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
```
Comment (DomT4) [inline Formula/plank.rb]: Can drop this now the stronger test exists.

File: Formula/plank.rb [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        "$schema": "http://json-schema.org/schema#",
+        "type": "object",
+        "properties": {
+          "id": { "type": "string" },
+          "link": { "type": "string", "format": "uri"}
+         }
+      }
+    EOS
+    system "#{bin}/plank", "--lang", "objc,flow", "--output_dir", testpath, "pin.json"
+    assert_predicate (testpath/"Pin.h"), :exist?, "[ObjC] Generated file does not exist"
```
Comment (DomT4) [inline Formula/plank.rb]: No need for the parens here. `assert_predicate testpath/"Pin.h", :exist?` etc should work just fine.

**Review by DomT4 — CHANGES_REQUESTED** [review]

**Review by DomT4 — APPROVED** [review]
This seems alright to me, FWIW, @ilovezfs.

File: Formula/plank.rb [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+  homepage "https://pinterest.github.io/plank/"
+  url "https://github.com/pinterest/plank/archive/v1.2.tar.gz"
+  sha256 "9ee8c248d7b4488f7f952b01d913b34cd83bd40111092fd49015d1c63b9e885d"
+  head "https://github.com/pinterest/plank.git"
+
+  depends_on :xcode => ["9.0", :build]
+
+  def install
+    system "make", "archive"
+    bin.install ".build/release/plank"
```
Comment (ilovezfs) [inline Formula/plank.rb]: please request a `make install` target upstream

File: Formula/plank.rb [hunk 9]
```diff
@@ -0,0 +1,33 @@
+class Plank < Formula
+  desc "Framework for generating immutable model objects"
+  homepage "https://pinterest.github.io/plank/"
+  url "https://github.com/pinterest/plank/archive/v1.2.tar.gz"
+  sha256 "9ee8c248d7b4488f7f952b01d913b34cd83bd40111092fd49015d1c63b9e885d"
+  head "https://github.com/pinterest/plank.git"
```
Comment (ilovezfs) [inline Formula/plank.rb]: let's leave this out

Comment (rahul-malik) [reply Formula/plank.rb]: Will remove.

File: Formula/plank.rb (hunk 8 above)
Comment (rahul-malik) [reply Formula/plank.rb]: Can you explain this a bit more? Using `bin.install` seemed like the correct way to install binaries from the documentation

Comment (ilovezfs) [reply Formula/plank.rb]: We don't want to have to keep track of what should or shouldn't be installed and where if it later changes. So `make install PREFIX=/usr/local/Cellar/plank/1.2` should be possible.

**Comment by rahul-malik** [comment]
@ilovezfs @DomT4 - I've released a new version of plank with a `make install` command and updated the formula here. Please let me know if there is anything else missing before merging

File: Formula/plank.rb (hunk 8 above)
Comment (rahul-malik) [reply Formula/plank.rb]: Thanks for explaining, it is addressed now

**Comment by rahul-malik** [comment]
@ilovezfs @DomT4 - Just checking in to see if there is anything else I need to do before this merges.

**Review by DomT4 — APPROVED** [review]
Seems alright to me at least. @ilovezfs is the one whose opinion you need at this point.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Review by ilovezfs — APPROVED** [review]

**Comment by rahul-malik** [comment]
@ilovezfs - Thanks for approving and getting this ready to merge!

**Comment by ilovezfs** [comment]
Thanks for your first contribution to Homebrew, @rahul-malik! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**